File: go-stream/send.go
package main

import (
	"bufio"
	"fmt"
	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/amqp"
	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/stream"
	"os"
)

func CheckErrSend(err error) {
	if err != nil {
		fmt.Printf("%s ", err)
		os.Exit(1)
	}
}
func main() {
	env, err := stream.NewEnvironment(
		stream.NewEnvironmentOptions().
			SetHost("localhost").
			SetPort(5552).
			SetUser("guest").
			SetPassword("guest"))
	CheckErrSend(err)

	streamName := "hello-go-stream"
	err = env.DeclareStream(streamName,
		&stream.StreamOptions{
			MaxLengthBytes: stream.ByteCapacity{}.GB(2),
		},
	)
	CheckErrSend(err)

	producer, err := env.NewProducer(streamName, stream.NewProducerOptions())
	CheckErrSend(err)

	// Send a message
	err = producer.Send(amqp.NewMessage([]byte("Hello world")))
	CheckErrSend(err)
	fmt.Printf(" [x] 'Hello world' Message sent\n")

	reader := bufio.NewReader(os.Stdin)
	fmt.Println(" [x] Press enter to close the producer")
	_, _ = reader.ReadString('\n')
	err = producer.Close()
	CheckErrSend(err)
}

File: python-stream/send.py
import asyncio

from rstream import Producer

STREAM_NAME = "hello-python-stream"
# 5GB
STREAM_RETENTION = 5000000000


async def send():
    async with Producer(
        host="localhost",
        username="guest",
        password="guest",
    ) as producer:
        await producer.create_stream(
            STREAM_NAME, exists_ok=True, arguments={"MaxLengthBytes": STREAM_RETENTION}
        )

        await producer.send(stream=STREAM_NAME, message=b"Hello, World!")

        print(" [x] Hello, World! message sent")

        input(" [x] Press Enter to close the producer  ...")


asyncio.run(send())
